gem/gem.py

```diff
@@ -782,6 +783,58 @@ def __new__(cls, i, j):
         return self
 
 
+class Inverse(Node):
+    """ Inverse is a shape to shape operation.
+
+        :arg aggregate: Tensor to be inversed
+        :arg multiindex: Indices for result of Inverse
+
+        Inverse is different from ComponentTensor, because its children must have shape.
+        It does no transformation between free indices and shape, but the multiindex is still necessary.
+        The multiindices of the children are used for indexing them when passing to inverse function call,
+        the multiindex of the Inverse node itself is used to index the result of the function call.
+    """
+    __slots__ = ('children', 'shape')
+    __back__ = ()
+
+    def __new__(cls, aggregate):
+        assert aggregate.shape
```

Comment:
This check seems pretty weak, only checks that the node is non-scalar.  You would expect square matrices, no?  Then write instead:
```python
assert len(aggregate.shape) == 2
assert aggregate.shape[0] == aggregate.shape[1]
```

Reply:
Also, why is it called `aggregate`? Just call it `matrix` or something... (if that's what you expect)